scalableminds/webknossos
# Refactor Antd Table columns

In preparation for upgrading to React 17 (#7765), I ran into trouble with Typescript and infinite recursions ("type definition to deep") resulting in crashes. Most likely this is not caused by the React update per se but some other dependency, likely any from the `rc-*` packages used by `antd`. 

In particular, Typescript was unhappy with the type `RowRenderer` in any of the table `<Column>` elements:
```
<Column
   ...
  render: (_name: string, rowRenderer: RowRenderer) => rowRenderer.renderNameColumn()
/>
```

Instead of using `<Column />` children for the `FixedExpandableTable`, I refactored this components to use objects and props instead, e.g. `<Table columns={ ... } />`. Judging from the [antd documentation](https://ant.design/components/table#api) this seems to be the preferred way these days. (Similar to `<Dropdown />`, `<Menu />` etc).

### Steps to test:
- Make sure that Dashboard > Datasets table works
- Make sure that Admin > Tasks table works

### Issues:
- Contributes to #7765 

------
(Please delete unneeded items, merge only when none are left open)
